test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -338,6 +342,14 @@ func (sm *StackManager) getStack(ctx context.Context, name string, config runner
 	progressStreamsUpOption := sm.getProgressStreamsOnUp(logger)
 	progressStreamsDestroyOption := sm.getProgressStreamsOnDestroy(logger)
 
+	//initialize datadog event sender
+	if datadodatadogEventSender == nil {
+		datadodatadogEventSender, err = newDatadogEventSender(logger)
+		if err != nil {
+			fmt.Fprintf(logger, "error when creating datadog event sender: %v\n", err)
+		}
+	}
```

Comment:
Would it make sense to move this initialisation in the `sendEventToDatadog` function to avoid potential errors?

Reply:
I could move it there, with a call to https://pkg.go.dev/sync#Once.Do